### PR TITLE
Implemented setting the pagination and sorting config in URL query parameters

### LINKED ISSUE #1767: Implement setting pagination and sorting config in URL query parameters
Refer: https://github.com/bigbinary/neeto-engineering-web/issues/413

### PR DESCRIPTION
**Pull request by Calvinwilson99**
- Fixes #1767 

**Description**

- Added: Mechanism to set the pagination and sorting config in URL query parameters for `Table`.

**Checklist**

- [x] I have made corresponding changes to the documentation.
- ~~[ ] I have updated the types definition of modified exports.~~
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary label (patch/minor/major - If package publish
      is required).

### CODE REVIEW

**Comment by Calvinwilson99** [comment]
@ajmaln _a. Please review.

Any `neetoUI` table in products and nanos that are not overriding the `onChange` prop will be affected by this change. We need to look into handling it during release.
If the `onChange` prop is being overriden the changes in this PR will not be functional, so we can change them after release.

**Review by ajmaln — COMMENTED** [review]
@Calvinwilson99 _a As discussed, let's add a boolean prop to enable/disable the behaviour

**Comment by Calvinwilson99** [comment]
@ajmaln _a. Added prop to disable setting query parameters in URL.

File: stories/Components/TableStoriesDocs/TableSortingDocs.mdx [hunk 1]
```diff
@@ -32,3 +32,13 @@ const columns = [
...
+
+- By default, the `onChange` function is utilized to set the pagination and sorting configuration like
+  current page, page size, sort column and sort order in the URL query parameters with the following keys:
+  - `sort_by`: The `dataIndex` of the particular column in `snake_case`.
+  - `order_by`: The sort order, whether `ascend`, `descend` or `undefined`.
+  - `page`: The current page number.
+  - `page_size`: The page size specified by the product.
+
+  If this functionality is not required simply override the `onChange` prop with the required function or
+  noop if no functionality is needed.
```
Comment (ajmaln) [inline stories/Components/TableStoriesDocs/TableSortingDocs.mdx:44]: Mention about `shouldEnableSortingConfig`

File: src/components/Table/index.jsx [hunk 2]
```diff
@@ -47,6 +51,7 @@ const Table = ({
   bordered = true,
   onColumnUpdate = noop,
   components = {},
+  shouldSetTableConfig = false,
```
Comment (ajmaln) [inline src/components/Table/index.jsx:54]:
```suggestion
  preserveTableStateInQuery = false,
```

**Review by ajmaln — APPROVED** [review]
@Calvinwilson99 _a Please check the minor suggestions. Please assign to @amaldinesh7 for further review

**Comment by Calvinwilson99** [comment]
@amaldinesh7 _a. Please review.

**Review by amaldinesh7 — CHANGES_REQUESTED** [review]
@Calvinwilson99 _a Minor suggestion. Rest LGTM.

File: src/components/Table/constants.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export const URL_SORT_ORDERS = { ascend: "asc", descend: "desc" };
+
+export const TABLE_SORT_ORDERS = { asc: "ascend", desc: "descend" };
```
[the remark this answers is not in the document]
Comment (Calvinwilson99) [reply src/components/Table/constants.js:3]: @amaldinesh7 _a.
The table requires `ascend` and `descend` to work. The query parameters will be used by products to send queries to the backend. So, if we directly set `ascend` and `descend` in the URL without converting them, we would have to convert them to `asc` and `desc` in every product.
I think the better option would be to use the converted forms in the URL and products can use them as is.

**Review by amaldinesh7 — APPROVED** [review]

**Pull request merged**